test/neural_networks/test_sampler_qnn.py

```diff
@@ -0,0 +1,279 @@
+from qiskit.utils import QuantumInstance, algorithm_globals
+
+from qiskit_machine_learning.neural_networks import CircuitQNN
+from qiskit_machine_learning.neural_networks.sampler_qnn import SamplerQNN
+import qiskit_machine_learning.optionals as _optionals
+
+algorithm_globals.random_seed = 42
+
+DEFAULT = "default"
+SHOTS = "shots"
+SPARSE = [True, False]
+SAMPLERS = [DEFAULT, SHOTS]
+INTERPRET_TYPES = [2]
+BATCH_SIZES = [1, 2]
+
+
+@ddt
+class TestSamplerQNN(QiskitMachineLearningTestCase):
+    """Sampler QNN Tests."""
+
+    def setUp(self):
+        super().setUp()
+        algorithm_globals.random_seed = 12345
+
+        # # define test circuit
+        # num_qubits = 3
+        # self.qc = RealAmplitudes(num_qubits, entanglement="linear", reps=1)
+
+        # define feature map and ansatz
+        num_qubits = 2
+        feature_map = ZZFeatureMap(num_qubits, reps=1)
+        var_form = RealAmplitudes(num_qubits, reps=1)
+
+        # construct circuit
+        self.qc = QuantumCircuit(num_qubits)
+        self.qc.append(feature_map, range(2))
+        self.qc.append(var_form, range(2))
+
+        # store params
+        self.input_params = list(feature_map.parameters)
+        self.weight_params = list(var_form.parameters)
+
+        # define interpret functions
+        def interpret_1d(x):
+            return sum((s == "1" for s in f"{x:0b}")) % 2
+
+        self.interpret_1d = interpret_1d
+        self.output_shape_1d = 2  # takes values in {0, 1}
+
+        def interpret_2d(x):
+            return np.array([self.interpret_1d(x), 2 * self.interpret_1d(x)])
+
+        self.interpret_2d = interpret_2d
+        self.output_shape_2d = (
+            2,
+            3,
+        )  # 1st dim. takes values in {0, 1} 2nd dim in {0, 1, 2}
+
+        # define sampler primitives
+        self.sampler = Sampler()
+        self.sampler_shots = Sampler(options={"shots": 100})
+
+    def _get_qnn(self, sparse, sampler_type, interpret_id):
+        """Construct QNN from configuration."""
+
+        # get quantum instance
+        if sampler_type == SHOTS:
+            sampler = self.sampler_shots
+        elif sampler_type == DEFAULT:
+            sampler = self.sampler
+        else:
+            sampler = None
+
+        # get interpret setting
+        interpret = None
+        output_shape = None
+        if interpret_id == 1:
+            interpret = self.interpret_1d
+            output_shape = self.output_shape_1d
+        elif interpret_id == 2:
+            interpret = self.interpret_2d
+            output_shape = self.output_shape_2d
+
+        # construct QNN
+        qnn = SamplerQNN(
+            sampler=sampler,
+            circuit=self.qc,
+            input_params=self.input_params,
+            weight_params=self.weight_params,
+            sparse=sparse,
+            interpret=interpret,
+            output_shape=output_shape,
+        )
+        return qnn
+
+    def _verify_qnn(
+        self,
+        qnn: CircuitQNN,
```

Comment:
I don;t think this should be checking using an entity which will go away. Ideally it should check against some known reference value(s) we expect the outcome to be, Also if doing multiple tests are these dependent on one another in sequence where the first one failing we expect the rest to fail. Or is there independence whereby these can be done using sub-test which is so can give a clearer picture of what might be happening if something fails since more tests may be run.